### DISCUSSION TITLE
Number of nodes blowing up when casadi's linspace is used with MX variables as start and end points

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3523] [Q&A] [answered]
I was using casadi's linspace in my application which caused the number of nodes in the problem to be significantly high despite the grid being depend only on the start and end points which in my casee happend to be symbolic MX scalars. Is this a bug ? 

![image](https://github.com/casadi/casadi/assets/16457676/d93f813d-1299-4188-b8eb-835b8fd46080)

![image](https://github.com/casadi/casadi/assets/16457676/059ef5ee-862d-4d56-a566-0c3a4acc7610)

As a workaround,I created a custom linspace for my application which resolved the issue. Hope this can be useful others too.

Here is a MWE.
```
import casadi as cs 
import matplotlib.pyplot as plt
# symbolic vars
x=cs.MX.sym('x',1,1)
y=cs.MX.sym('y',1,1)
# nodes based on linspace and custom method
node_lin=[]
node_cus=[]
N=[]

for i in range(1000):
    N.append(i+2)
    node_lin.append((cs.n_nodes(cs.linspace(x,y,i+2+1))))
    node_cus.append((cs.n_nodes(x*(1-cs.linspace(0,i+2,i+2+1)/(i+2))+y*cs.linspace(0,i+2,i+2+1)/(i+2))))

plt.figure
plt.plot(cs.hcat(N).full(),cs.hcat(node_lin).full(),'o-')   
plt.plot(cs.hcat(N).full(),cs.hcat(node_cus).full(),'*-') 
plt.grid(True)
plt.show()

#check for correctness in implementation
x=5
y=20
err=[]

for i in range(1000):
    err.append((cs.sumsqr(cs.linspace(x,y,i+2+1)-(x*(1-cs.linspace(0,i+2,i+2+1)/(i+2))+y*cs.linspace(0,i+2,i+2+1)/(i+2)))))

plt.figure
plt.plot(cs.hcat(N).full(),cs.hcat(err).full(),'o-')
plt.grid(True)
plt.show()
```

**Comment by jgillis** [comment, marked as answer]
You're right, this is a missed opportunity to specialize `linspace` for MX. I've spawned an issue #3531
